### PR TITLE
Fixing #801

### PR DESCRIPTION
**Pull request by mbad0la**
@mariobehling current changes make the pop-over design look like this.
openappissue1

Please suggest further steps and/or any changes to this.

### CODE REVIEW

**Comment by mariobehling** [comment]
@mbad0la Thank you. Could you check, if you could reduce the number of hound-ci issues, please?

**Comment by mbad0la** [comment]
Yeah @mariobehling I was hoping to resolve all the lint issues once the design was taken care of. Are the current design changes fair enough?

**Comment by mariobehling** [comment]
Thanks. Looks good to me as far as I can see. Cheers.

**Comment by mbad0la** [comment]
@aayusharora @mariobehling please review and merge.

File: src/backend/_scss/application.scss [hunk 1]
```diff
@@ -522,39 +522,63 @@ a {
...
+          font-weight: 400;
+          color: $gray-dark;
+        }
     }
 
     .pop-footer {
       background: $white;
       border-top-left-radius: 5px;
       border-top-right-radius: 5px;
+      background: #fff;
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: Use variable color from config.scss

File: src/backend/_scss/application.scss [hunk 2]
```diff
@@ -522,39 +522,63 @@ a {
...
+          margin-left: -5px;
+        }
+
+        .pop-over>.col-xs-12, .pop-over>.col-md-12 {
+          padding: 0px;
+        }
+
+        .pop-over {
+          padding: 0px;
+          margin: 0px;
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: Please do this.

File: src/backend/_scss/application.scss [hunk 3]
```diff
@@ -522,39 +522,63 @@ a {
...
+    .pop {
+      background-color: #fff;
+      margin-left: 20px;
+
+        a {
+          margin-left: -5px;
+        }
+
+        .pop-over>.col-xs-12, .pop-over>.col-md-12 {
+          padding: 0px;
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: Please do this

File: src/backend/_scss/application.scss [hunk 4]
```diff
@@ -522,39 +522,63 @@ a {
...
+
+    .pop {
+      background-color: #fff;
+      margin-left: 20px;
+
+        a {
+          margin-left: -5px;
+        }
+
+        .pop-over>.col-xs-12, .pop-over>.col-md-12 {
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: Don't refer col-md-12 like this. Instead make a class and refer like .pop-over>classname

File: src/backend/_scss/application.scss [hunk 5]
```diff
@@ -522,39 +522,63 @@ a {
...
+    .pop>h4 {
+      margin-left: -20px;
+    }
+
+    .pop>.pop-over.row {
+      padding: 10px 20px 0px 0px;
+    }
+
+    .pop {
+      background-color: #fff;
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: Please do this from config.scss

File: src/backend/_scss/application.scss [hunk 6]
```diff
@@ -522,39 +522,63 @@ a {
...
-      word-wrap: break-word;
+    .pop>h4 {
+      margin-left: -20px;
+    }
+
+    .pop>.pop-over.row {
+      padding: 10px 20px 0px 0px;
+    }
+
+    .pop {
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: Please do this

File: src/backend/_scss/application.scss [hunk 7]
```diff
@@ -522,39 +522,63 @@ a {
...
-      font-weight: 400;
-      line-height: 20px;
-      margin: 10px;
-      padding: 2%;
-      word-wrap: break-word;
+    .pop>h4 {
+      margin-left: -20px;
+    }
+
+    .pop>.pop-over.row {
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: Please do this

File: src/backend/_scss/application.scss [hunk 8]
```diff
@@ -522,39 +522,63 @@ a {
...
+    }
 
-    .pop-over {
-      color: $gray-dark;
-      font-weight: 400;
-      line-height: 20px;
-      margin: 10px;
-      padding: 2%;
-      word-wrap: break-word;
+    .pop>h4 {
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: Please do this

File: src/backend/_scss/application.scss [hunk 9]
```diff
@@ -522,39 +522,63 @@ a {
   min-height: 255px;
   position: absolute;
   transition: all 0.5s ease;
+  background-color:white;
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: white is not allowed instead use $white. Here background-color seems over-written from 518

Comment (mbad0la) [inline src/backend/_scss/application.scss]: Everything has been sorted out @aayusharora , I don't have any more issues in scss linting. Check the latest status please.

File: src/backend/_scss/application.scss [hunk 10]
```diff
@@ -525,35 +527,56 @@ a {
...
+
   .pop {
     background-color: $white;
+    margin-left: 20px;
+
+    a {
+      margin-left: -5px;
+    }
+
+    .pop-over>.col-xs-12,
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: .pop-over>.col-xs-12, Use class instead mentioning bootstrap columns.

Comment (mbad0la) [inline src/backend/_scss/application.scss]: Done @aayusharora . Please review.

File: src/backend/_scss/application.scss [hunk 11]
```diff
@@ -525,35 +527,56 @@ a {
...
-      padding: 5px 5px 0 20px;
-    }
+  .pop-header {
+    background: $span-background;
+    border-bottom: 1px solid $gray-perfect;
+    border-top-left-radius: 5px;
+    border-top-right-radius: 5px;
+    height: 35px;
+    margin-left: -20px;
+    padding-bottom: 0;
```
Comment (mbad0la) [inline src/backend/_scss/application.scss]: @aayusharora The scss linter pointed out that the shorthand can't have 4 arguments hence I had to put them in separately. Is there a different way to do it?

Comment (aayusharora) [inline src/backend/_scss/application.scss]: Is this giving lint error :: Example
> padding : 10px 0;

Comment (mbad0la) [inline src/backend/_scss/application.scss]: If I recall correctly `padding: 10px 0;` will assign 10px padding to top and bottom while 0px padding to left and right. As the padding in my solution doesn't show a behaviour that can fit any particular valid shorthand, I had to write them separately. Am I wrong @aayusharora ?

**Comment by mariobehling** [comment]
@mbad0la It would be great to merge this PR. Could you fix the remaining points, please? I think we are close to merge. So, keep it up!

File: src/backend/_scss/application.scss [hunk 12]
```diff
@@ -416,7 +416,9 @@ a {
   box-shadow: 0 1px 2px $event-box-shadow;
   cursor: pointer;
   display: block;
-  margin: 0 12px 12px 0;
+  margin: 12px;
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: Why this is overwritten by the three lines @mbad0la , I was considering shorthand like these :)

Comment (mbad0la) [inline src/backend/_scss/application.scss]: @aayusharora  `margin: 0 12px 12px 0;` was giving linting errors.

File: src/backend/_scss/application.scss (hunk 11 above)
Comment (aayusharora) [inline src/backend/_scss/application.scss]: Can we change line 419 to margin: 0 12px 12px 0; ? to replace line 420 and 421

File: src/backend/_scss/application.scss [hunk 13]
```diff
@@ -522,63 +524,59 @@ a {
...
-      padding: 10px 20px 0px 0px;
-    }
+  .pop>h4 {
+    margin-left: -20px;
+  }
 
-    .pop {
-      background-color: #fff;
-      margin-left: 20px;
+  .pop>.pop-over.row {
```
Comment (aayusharora) [inline src/backend/_scss/application.scss]: This can be written better :)

Comment (mbad0la) [inline src/backend/_scss/application.scss]: How so? Please guide.

File: src/backend/_scss/application.scss (hunk 11 above)
Comment (aayusharora) [inline src/backend/_scss/application.scss]: Can we replace line 540 to 544 by shorthand.

Comment (mbad0la) [inline src/backend/_scss/application.scss]: Same issue. They were initially in shorthand, but were giving lint errors. :(

Comment (aayusharora) [inline src/backend/_scss/application.scss]: Now, I got this, have to edit hound.yml  :) Thanks

**Comment by aayusharora** [comment]
Thanks @mbad0la for this. We have to edit hound.yml to have shorthands.

**Comment by mbad0la** [comment]
@aayusharora thanks :)

Although I do recommend moving away from hound and putting up lint tests in our travis build itself. What do you think? Hound ends up spamming on the PR threads making conversations cumbersome to follow. Not to mention, Travis is well-equiped to handle lint tests as well, so using a different service is a redundant addition to the dev-pipeline.

**Pull request merged**